explorer/server.go

```diff
@@ -0,0 +1,210 @@
+package explorer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/gomodule/redigo/redis"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zos/client"
+	"github.com/threefoldtech/zos/pkg/rmb"
+)
+
+func (a *App) listFarms(w http.ResponseWriter, r *http.Request) {
+	r, err := a.handleRequestsQueryParams(r)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(http.StatusText(http.StatusBadRequest)))
+		return
+	}
+	maxResult, pageOffset := getMaxResult(r.Context()), getOffset(r.Context())
+
+	queryString := fmt.Sprintf(`
+	{
+		farms (limit:%d,offset:%d) {
+			name
+			farmId
+			twinId
+			version
+			farmId
+			pricingPolicyId
+		}
+		publicIps{
+			id
+			ip
+			farmId
+			contractId
+			gateway
+			
+		}
+	}
+	`, maxResult, pageOffset)
+
+	_, err = queryProxy(queryString, a.explorer, w)
+
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(http.StatusText(http.StatusBadRequest)))
+	}
+}
+
+func (a *App) listNodes(w http.ResponseWriter, r *http.Request) {
+	r, err := a.handleRequestsQueryParams(r)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(http.StatusText(http.StatusBadRequest)))
+	}
+
+	maxResult := getMaxResult(r.Context())
+	pageOffset := getOffset(r.Context())
+	isSpecificFarm := getSpecificFarm(r.Context())
+
+	queryString := fmt.Sprintf(`
+	{
+		nodes(limit:%d,offset:%d,%s){
+			version          
+			id
+			nodeId        
+			farmId          
+			twinId          
+			country
+			gridVersion  
+			city         
+			uptime           
+			created          
+			farmingPolicyId
+			updatedAt
+			cru
+			mru
+			sru
+			hru
+		publicConfig{
+			gw4
+			ipv4
+			ipv6
+			gw6
+		  }
+		}
+	}
+	`, maxResult, pageOffset, isSpecificFarm)
+
```

Comment:
the content type `w.Header().Add("Content-Type", "application/json")` here and `listFarms` can be specified like in getNode